lib/record-events-browser.js

```diff
@@ -69,11 +69,20 @@ export function recordEvent(eventCollectionOrConfigObject, eventBody, callback){
   }
 
   this.emit('recordEvent', eventCollection, extendedEventsHash);
-
+  
   if (!Keen.enabled) {
     handleValidationError.call(this, 'Keen.enabled is set to false.', callback);
     return false;
   }
+  if (Keen.optedOut) {
+    handleValidationError.call(this, 'Keen.optedOut is set to true.', callback);
+    return false;
+  }
```

Comment:
with specific configuration for autocollector (custom .catch) it throws an error. Let's change this code to the one below. this way we'll base our responses more on Promises.

```javascript
  if (Keen.optedOut) {
    return Promise.resolve({
      created: false,
      message: 'Keen.optedOut is set to true.'
    });
  }
```